src/HTTPFile.cc

```diff
@@ -93,6 +94,41 @@ int parse_path(const std::string &storagePrefixStr, const char *pathStr,
 }
 
 int HTTPFile::Open(const char *path, int Oflag, mode_t Mode, XrdOucEnv &env) {
+	if (m_is_open) {
+		m_log.Log(LogMask::Warning, "HTTPFile::Open",
+				  "File already open:", path);
+		return -EBADF;
+	}
+	if (Oflag & O_CREAT) {
+		m_log.Log(LogMask::Info, "HTTPFile::Open",
+				  "File opened for creation:", path);
+	}
+	if (Oflag & O_APPEND) {
+		m_log.Log(LogMask::Info, "HTTPFile::Open",
+				  "File opened for append:", path);
+	}
+	if (Oflag & (O_RDWR | O_WRONLY)) {
+		m_write = true;
+		m_log.Log(LogMask::Debug, "HTTPFile::Open",
+				  "File opened for writing:", path);
+		m_write_lk.reset(new std::mutex);
+	}
+	// get the expected file size; only relevant for O_RDWR | O_WRONLY
+	char *asize_char;
+	if ((asize_char = env.Get("oss.asize"))) {
+		off_t result{0};
+		auto [ptr, ec] = std::from_chars(
+			asize_char, asize_char + strlen(asize_char), result);
+		if (ec == std::errc()) {
+			m_object_size = result;
+		} else {
+			std::stringstream ss;
+			ss << "Opened file has oss.asize set to an unparseable value: "
+			   << asize_char;
+			m_log.Log(LogMask::Warning, "HTTPFile::Open", ss.str().c_str());
```

Comment:
Turn this to a hard error - fail the file open.

Similarly, check to see if a negative value was parsed and reject that as well.